blog/12-12-2022-advance-hardhat-configration-on-celo-using-plugins/advance-hardhat-configration-on-celo-using-plugins.md

```diff
@@ -0,0 +1,751 @@
+
+*Note:* Hardhat components are files/folders within Hardhat context i.e. within the scope of its reach, such as “test” folder. 
+
+## @nomicfoundation/hardhat-toolbox
+
+A set of coordinated tools developed and recommended by the hardhat officials that bundles the commonly used packages and plugins. When you initialize a hardhat project, “@nomicfoundation/hardhat-toolbox” will be the default umbrella package that houses other tools such as “@nomiclabs/hardhat-etherscan”, “@nomiclabs/hardhat-ethers” etc. Importing the library alongside the pre-installed tools will only make them redundant. 
+
+Even though you can do a lot with this plugin, yet you’re limited to the extent of its scope. When you need to make use of other plugins, it is advisable to remove  “@nomicfoundation/hardhat-toolbox” from the configuration before proceeding to avoid dependency conflicts. 
+
+```bash
+@nomicfoundation/hardhat-chai-matchers
+
+```
+
+This is a Hardhat component that blends Ethereum-specific functionalities with the Chai assertion library to make testing smart contracts more easy, readable, and concise. The recent version of Hardhat comes with “@nomicfoundation/hardhat-toolbox” so you do not need to install it. If you like to set things up independently, please continue with this section.
+
+This plugin depends on other libraries or packages to function as intended. You will install it as a peer/dev dependency along with the peer packages if you’re using an older version of Node.
+
+Peer dependencies
+  - chai 
+  - @nomiclabs/hardhat-ethers 
+  - ethers
+
+```bash
+npm install --save-dev @nomicfoundation/hardhat-chai-matchers chai @nomiclabs/hardhat-ethers ethers
+```
+Or
+
+```bash
+
+yarn add --dev @nomicfoundation/hardhat-chai-matchers chai @nomiclabs/hardhat-ethers ethers
+
+```
+
+Include it in the Hardhat config file:
+
+```bash “js”
+require("@nomicfoundation/hardhat-chai-matchers")
+
+```
+
+```bash “typescript”
+import "@nomicfoundation/hardhat-chai-matchers";
+
+```
+
+
+## @openzeppelin/hardhat-upgrades
+
+This tool reduces your tasks when writing upgradeable contracts by modifying Hardhat scripts adding new functions that enable you to deploy and manage proxied contracts, so anytime you release a new version of your dApp, you only need a minimal command to update it, and your users can start interacting with the newer version.
+If you’re looking for a tutorial on how to write upgradeable contracts, [this blogpost](https://docs.celo.org/blog/tutorials/how-to-create-an-upgradeable-smart-contract-in-celo) might be helpful.
+
+Openzeppelin/hardhat-upgrades depend on etherjs library as peer dependency to function properly. We will install it as a dev dependency. alongside the peer.
+
```

Comment:
This tool reduces your tasks when writing upgradeable contracts by modifying Hardhat scripts and adding new functions that enable you to deploy and manage proxied contracts, so anytime you release a new version of your dApp, you only need a minimal command to update it, and your users can start interacting with the newer version.
		If you’re looking for a tutorial on how to write upgradeable contracts, [this blog post](https://docs.celo.org/blog/tutorials/how-to-create-an-upgradeable-smart-contract-in-celo) might be helpful.
		
		Openzeppelin/hardhat-upgrades depend on etherjs library as peer dependency to function properly. We will install it as a dev dependency. alongside the peer.